app/helpers/katello/concerns/settings_helper_extensions.rb

```diff
@@ -0,0 +1,45 @@
+module Katello
+  module Concerns
+    module SettingsHelperExtensions
+      extend ActiveSupport::Concern
+
+      included do
+        alias_method_chain :value, :katello
+      end
+
+      def value_with_katello(setting)
+        return value_without_katello(setting) unless [
+          'default_download_policy',
+          'katello_default_finish',
+          'katello_default_iPXE',
+          'katello_default_provision',
+          'katello_default_ptable',
+          'katello_default_PXELinux',
+          'katello_default_user_data'].include?(setting.name)
+
+        case setting.name
+        when "default_download_policy"
+          edit_select(setting, :value, :select_values => {:immediate => "immediate", :on_demand => "on_demand", :background => "background"}.to_json)
+        when "katello_default_finish"
+          edit_select(setting, :value, :select_values => katello_template_setting_values(5))
+        when "katello_default_iPXE"
+          edit_select(setting, :value, :select_values => katello_template_setting_values(3))
+        when "katello_default_provision"
+          edit_select(setting, :value, :select_values => katello_template_setting_values(4))
+        when "katello_default_ptable"
+          edit_select(setting, :value, :select_values => Hash[Template.all.where(:type => "Ptable").map { |tmp| [tmp[:name], tmp[:name]] }].to_json)
+        when "katello_default_PXELinux"
+          edit_select(setting, :value, :select_values => katello_template_setting_values(1))
+        when "katello_default_user_data"
+          edit_select(setting, :value, :select_values => katello_template_setting_values(7))
+        end
+      end
+
+      private
+
+      def katello_template_setting_values(id)
+        Hash[Template.all.where(:type => "ProvisioningTemplate").where(:template_kind_id => id).map { |tmp| [tmp[:name], tmp[:name]] }].to_json
```

Comment:
We can do it that way if you like but i feel it's unnecessary as the id are unlikely to change and using names means we have to query to DB more?